### PR TITLE
Add/Update push Javadoc

### PR DESCRIPTION
**Pull request by danielpassos**

### CODE REVIEW

File: push/src/main/java/org/aerogear/mobile/push/PushService.java [hunk 1]
```diff
@@ -93,10 +96,21 @@ public boolean requiresConfiguration() {
...
+     */
     public void registerDevice(final Callback callback) {
         registerDevice(new UnifiedPushConfig(), callback);
     }
 
+    /**
+     * Register the device on Unified Push Server
+     *
+     * @param unifiedPushConfig Unified Push configuration to be send to the Unified Push Server
+     * @param callback A callback to handler success or fail registration
```
Comment (wtrocki) [inline push/src/main/java/org/aerogear/mobile/push/PushService.java]: `handle`

File: push/src/main/java/org/aerogear/mobile/push/PushService.java [hunk 2]
```diff
@@ -160,10 +174,21 @@ public void registerDevice(final UnifiedPushConfig unifiedPushConfig, final Call
 
     }
 
+    /**
+     * Unregister the device on Unified Push Server
+     *
+     * @param callback A callback to handler success or fail unregister
```
Comment (wtrocki) [inline push/src/main/java/org/aerogear/mobile/push/PushService.java]: A callback to `handle` success or fail unregistration

File: push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java [hunk 3]
```diff
@@ -6,26 +6,54 @@
...
     public String getSender() {
         return sender;
     }
 
+    /**
+     * Firebase sender Id registered for this application.
+     *
+     * @param sender the new senderId
+     */
     public void setSender(String sender) {
```
Comment (wtrocki) [inline push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java]: Let's rename this to senderId then :P

File: push/src/main/java/org/aerogear/mobile/push/UnifiedPushConfig.java [hunk 4]
```diff
@@ -17,18 +17,40 @@
     private String alias;
     private List<String> categories = Collections.emptyList();
 
+    /**
+     * The Alias is an identifier of the user of the system.
```
Comment (secondsun) [inline push/src/main/java/org/aerogear/mobile/push/UnifiedPushConfig.java]: "Alias" should be lower case

File: push/src/main/java/org/aerogear/mobile/push/UnifiedPushConfig.java [hunk 5]
```diff
@@ -17,18 +17,40 @@
...
+     *
+     * @return alias
+     *
+     */
     public String getAlias() {
         return alias;
     }
 
+    /**
+     * The Alias is an identifier of the user of the system.
```
Comment (secondsun) [inline push/src/main/java/org/aerogear/mobile/push/UnifiedPushConfig.java]: "Alias" should be lower case

File: push/src/main/java/org/aerogear/mobile/push/PushService.java [hunk 6]
```diff
@@ -203,9 +228,16 @@ public void unregisterDevice(final UnifiedPushConfig unifiedPushConfig,
 
     }
 
-    private String getHashedAuth(final String username, final char[] password) {
+    /**
+     * Provide an Auth hash to be used to authenticate in Unified Push Service
```
Comment (secondsun) [inline push/src/main/java/org/aerogear/mobile/push/PushService.java]: Auth should be lower case here and in the return.

File: push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java [hunk 7]
```diff
@@ -4,30 +4,58 @@
 
     private String variant;
     private String secret;
-    private String sender;
+    private String senderId;
 
+    /**
+     * ID of the Variant from the AeroGear UnifiedPush Server.
```
Comment (secondsun) [inline push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java]: lowercase variant

File: push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java [hunk 8]
```diff
@@ -4,30 +4,58 @@
...
+     *
+     * @return the current variant id
+     */
     public String getVariant() {
         return variant;
     }
 
+    /**
+     * ID of the Variant from the AeroGear UnifiedPush Server.
```
Comment (secondsun) [inline push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java]: lowercase variant

File: push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java [hunk 9]
```diff
@@ -4,30 +4,58 @@
...
+     *
+     * @param variant the new variantID
+     */
     public void setVariant(String variant) {
         this.variant = variant;
     }
 
+    /**
+     * Secret of the Variant from the AeroGear UnifiedPush Server.
```
Comment (secondsun) [inline push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java]: lowercase variant,  Would read better as "Secret for the variant from..."

File: push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java [hunk 10]
```diff
@@ -4,30 +4,58 @@
...
+     *
+     * @return the current Secret
+     */
     public String getSecret() {
         return secret;
     }
 
+    /**
+     * Secret of the Variant from the AeroGear UnifiedPush Server.
```
Comment (secondsun) [inline push/src/main/java/org/aerogear/mobile/push/UnifiedPushCredentials.java]: lowercase variant,  Would read better as "Secret for the variant from..."

**Pull request merged**